woocommerce/woocommerce
# Highlight main block at hover

> [!WARNING]  
> This PR is blocked by #47322 and #47356.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:






https://github.com/woocommerce/woocommerce/assets/4463174/c6eb0c74-aea5-4147-af2a-509b5d4f9455



Closes https://github.com/woocommerce/woocommerce/issues/47371.


This PR allows the user to highlight the main block with the hover. Also, it improves how the block toolbar is rendered. However, there are some leftovers that will be fixed in dedicated PRs:
- https://github.com/woocommerce/woocommerce/issues/47453
- https://github.com/woocommerce/woocommerce/issues/47454
- https://github.com/woocommerce/woocommerce/issues/47455
- https://github.com/woocommerce/woocommerce/issues/47456



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Ensure that you are using Gutenberg last trunk version.
2. Make sure you have the WooCommerce beta tester plugin installed
3. Head over to Tools > WCA Test Helper > Features
4. Make sure you see the `pattern-toolkit-full-composability` on the list.
5. Enable it.
6. Head over to WooCommerce > Home > Customize your store.
7. Click on "Start designing" and proceed to the Pattern Assembler.
8. Hover blocks and ensure that the main parent block is highlighted.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Highlight the pattern when the user hovers it.


#### Comment 


</details>
